snapcraft/parts/lifecycle.py

```diff
@@ -413,7 +413,7 @@ def _generate_metadata(
     snap_yaml.write(
         project,
         lifecycle.prime_dir,
-        arch=lifecycle.target_arch,
+        arch=project.get_build_for(),
         arch_triplet=lifecycle.target_arch_triplet,
```

Comment:
I'd like to understand this a bit more.

If we're building for multiple architectures (not arch-independent), I presume this gets generated on a per-arch basis? (If so that's fine.)

If we're building an architecture-independent snap, does an `arch` value get written to the `snap.yaml` file here? If so, this *could* be a problem if something ends up using that value for any reason.

Reply:
ping @sergiusens

Reply:
As discussed offline, I've fixed this behavior.